src/axom/inlet/LuaReader.hpp

```diff
@@ -139,23 +134,74 @@ class LuaReader : public Reader
+   *****************************************************************************
+   */
+  bool getIntMap(const std::string& id, std::unordered_map<int, int>& values);
+
+   /*!
+   *****************************************************************************
+   * \brief Get an index-double mapping for the given Lua array
+   *
+   * This performs any necessary retrieval and mapping from the given identifier
+   * to what is in the input file.
+   *
+   * \param [in]  id    The identifier to the string that will be retrieved
+   * \param [out] map The values of the doubles that were retrieved
+   *
+   * \return true if the array was able to be retrieved from the file
+   *****************************************************************************
+   */
+  bool getDoubleMap(const std::string& id, std::unordered_map<int, double>& values);
+
+   /*!
+   *****************************************************************************
+   * \brief Get an index-bool mapping for the given Lua array
+   *
+   * This performs any necessary retrieval and mapping from the given identifier
+   * to what is in the input file.
+   *
+   * \param [in]  id    The identifier to the string that will be retrieved
+   * \param [out] map The values of the bools that were retrieved
+   *
+   * \return true if the array was able to be retrieved from the file
+   *****************************************************************************
+   */
+  bool getBoolMap(const std::string& id, std::unordered_map<int, bool>& values);
+
+   /*!
+   *****************************************************************************
+   * \brief Get an index-string mapping for the given Lua array
    *
-   * This performs any necessary retrieval and mapping from the given fully
-   * qualified Sidre path name to what is in the input file.
+   * This performs any necessary retrieval and mapping from the given identifier
+   * to what is in the input file.
    *
-   * \param [in] id The path name to the bool that will be retrieved
+   * \param [in]  id    The identifier to the string that will be retrieved
+   * \param [out] map The values of the strings that were retrieved
    *
-   * \return true if Lua variable was found and Lua state was moved to
-   * variable
+   * \return true if the array was able to be retrieved from the file
    *****************************************************************************
    */
-  bool findVariable(const std::string& id);
+  bool getStringMap(const std::string& id, std::unordered_map<int, std::string>& values);
+private:
+  // Expect this to be called for only Inlet-supported types.
+  template<typename T>
+  bool getValue(const std::string& id, T& value);
 
-  lua_State* m_luaState;
+  // Expect this to be called for only Inlet-supported types.
+  template <typename T>
+  bool getMap(const std::string& id, std::unordered_map<int, T>& values, sol::type type);
+  sol::state m_lua;
```

Comment:
Suggestion (feel free to ignore):
For readability, I like to separate private methods from private variables with an extra access modifier, e.g.

```cpp
private:
  // Expect this to be called for only Inlet-supported types.
  template<typename T>
  bool getValue(const std::string& id, T& value);

  // Expect this to be called for only Inlet-supported types.
  template <typename T>
  bool getMap(const std::string& id, std::unordered_map<int, T>& values, sol::type type);

private:
  sol::state m_lua;
```

Reply:
@aeshapar can you do this or just put a line break between the member functions and variables?